src/lib/provable/test/arithmetic.unit-test.ts

```diff
@@ -45,7 +60,20 @@ equivalent({
   }
 );
 
-await equivalentAsync({ from: [field], to: divMod32Output }, { runs: 3 })(
+equivalent({
+  from: [field],
+  to: divModOutput,
+})(
+  (x) => {
+    assert(x < 1n << 128n, `x needs to fit in 128bit, but got ${x}`);
```

Comment:
This assertion fails because the random field is larger than 2^128. It works for divMod32 above, which is smaller, and that should fail first. Any ideas, @mitschabaude?

Reply:
the test checks whether the two functions are equivalent.
so the **test succeeds if both functions throw an error** for the same input.

so, it's not necessarily the case that "divMod32 should fail first". Maybe, `divMod32()` correctly raises an error if the input is larger than 2^64.

if this test fails, my idea would be to check why `divMod64()` doesn't throw an error on inputs larger than 2^128

Reply:
I checked both functions, and they throw an error on inputs larger than 2^128. The error disappeared when I either removed the `quotientBits === 64` case in `divMod64` or replaced it with a `rangeCheckN()`.

Reply:
I know what happens now, rangeCheck64 doesn't throw an error in constraint generation mode because snarky doesn't support checking custom gates. 

This isn't a problem for real proofs (because there, the proof would fail even if constraint checking succeeded).

So, I recommend to resolve this by simply not testing large field elements in the test, just make the inputs uint64